lib/ruby_lsp/internal.rb

```diff
@@ -1,6 +1,11 @@
 # typed: strict
 # frozen_string_literal: true
 
+# If YARP is in the bundle, we have to remove it from the $LOAD_PATH because it contains a default export named `prism`
+# that will conflict with the actual Prism gem
+yarp_require_paths = Gem.loaded_specs["yarp"]&.full_require_paths
+$LOAD_PATH.delete_if { |path| yarp_require_paths.include?(path) } if yarp_require_paths
+
```

Comment:
Not exactly. YARP v0.13 has two default export: `prism` and `yarp`.

If both YARP and Prism are in the bundle, the **Ruby LSP itself** loads the wrong one when we do `require "prism"`. I don't know why bundler picks YARPs' prism export.

<img width="933" alt="Screenshot 2023-10-18 at 1 56 19 PM" src="https://github.com/Shopify/ruby-lsp/assets/18742907/ce7ee68c-2337-4b00-98b7-901d44bfae1f">

Reply:
Ah ok, that makes more sense, thank you. Given that explanation, I agree that `Prism` would not already be loaded and just removing `yarp` require paths from `$LOAD_PATH` should be enough. Thanks for the explainer.

The reason why Bundler picks (actually it is not Bundler picking) `prism.rb` inside the `yarp/lib` folder is because that folder is listed before `prism/lib` folder in `$LOAD_PATH`. I think that would happen if `yarp` is listed before `ruby-lsp` in the Gemfile. Basically it is an ordering problem for when a gem's require paths are added to `$LOAD_PATH`.